### DISCUSSION TITLE
How do I use a firestore "map" type?

### DISCUSSION THREAD

**Discussion by KCErb** [tylim88/FirelordJS#97] [Q&A] [answered]
Hi there, I'm new to firebase/firestore/firelordJS :wave: I love this project, thank you for all of your hard work!

I woud like to use a map type in my "group" document. This is a group of users. So the "participants" field is an array of user ids. I want each user to be able to have their own label for the group.

For example, if my Mom and me are in a group. Her view of the group will have the label "KC"  and my view of the group will have the label "Mom". I'm implementing this with a `map` type in firestore and just learned that on the javascript side it will be a plain object, not an es6 Map.

So now I just need to make the MetaType

```js
export type Group = MetaTypeCreator<
  {
    labels: { [uid: string]: string },
    participants: string[],
  },
  'groupsCollection'
>
```

This says "I don't know what the key will be (since it'll be a user id) but I do know it'll be a string and I know the value will be a string too.

But now when I try to add a doc

```js
const groupDocRef = await addDoc(group.collection(), {
  labels: {[user.uid]: "Mom", [mom.uid]: "KC"},
  participants: [user.uid, mom.uid],
});
```

I get a warning "not assignable to type never"

<img width="580" alt="image" src="https://github.com/tylim88/FirelordJS/assets/4992244/b16e6591-2dec-4c2a-a665-9c7e0cbebc94">

It looks like the `write` property sets this as never

<img width="179" alt="image" src="https://github.com/tylim88/FirelordJS/assets/4992244/c803c6f4-3610-48c5-95ad-1a1e72293827">

How can I create a meta type for a map of `docId` to `string` using FirestoreJS?

**Comment by tylim88** [comment, marked as answer]
Hi there, thanks for using firelord

I am not able to reproduce your problem https://codesandbox.io/s/test-firelord-z9brsl?file=/src/testMap.ts

![image](https://github.com/tylim88/FirelordJS/assets/5227509/572b7f4f-dab1-49d7-905a-9a82ccdf7b46)


![image](https://github.com/tylim88/FirelordJS/assets/5227509/b97fcfd8-49ed-4e64-be6e-364cd4ce81ee)

make sure you install everything mentioned here, including `@firebase/rules-unit-testing` (as dev dependency)

the latest firelordjs version is 2.6.11

if the problem persist, please attach your repo link

**Reply by KCErb** [reply]
Yup, that was it. My typescript, firebase, and rules-unit-testing had all gotten way behind. Thank you!
